woocommerce/woocommerce-ios
# [Free Trial] Redesign Hub Menu

Closes: #9379 

# Why

This PR updates the Hub Menu to be redesigned from a Grid to a List.

**Note: This PR is on the bigger side(Sorry 😇 ) but It is because I needed to redesign it in one go, as doing it via feature flags was getting a bit too complex**

# How

- Most of the changes are in the `HubMenu` file. Here, the main changes are:
   - Update the main structure to use a `List`
   - Create a `Row` type to be used inside the `List`. This is a simpler version of the previous `TopBar` type.
   - Remove unneeded code & move the tap handling code to its own function.

- Remove `HubMenuElement` type. No longer used. It was the previous menu button.

- Update the `HubMenuViewModel` to provide two sections(Settings & Generals) instead of one.

- Update the `HubMenuItem` to add descriptions.

# Screenshots

Light | Dark | Big Fonts
--- | --- | ---
<img width="454" alt="Light" src="https://user-images.githubusercontent.com/562080/231657141-e49c217e-7fa1-4c59-9109-2b05efada530.png"> | <img width="444" alt="dark" src="https://user-images.githubusercontent.com/562080/231657135-91f0c7d1-6d52-4a88-87d5-b099f5c748cb.png"> | <img width="448" alt="big-fonts" src="https://user-images.githubusercontent.com/562080/231657149-c95c8e0d-566e-4e7c-a65b-a3c8b8ed58d9.png">

# Testing Steps

- Launch the app and navigate to the Menu Tab
- See the new view redesigned as a list
- Tap all buttons and make sure they navigate to the correct place.

---
